eng/common/scripts/Generate-PR-Diff.ps1

```diff
@@ -0,0 +1,51 @@
+<#
+.SYNOPSIS
+Script used to generate the diff.json file for a PR. Explicitly intended to work in a PR context.
+
+.DESCRIPTION
+Combines the result of git diff, some parsed details from the diff, and the PR number into a single JSON file. This JSON file is intended for use further along the pipeline.
+
+.PARAMETER ArtifactPath
+The folder in which the result will be written.
+
+.PARAMETER TargetPath
+The path under which changes will be detected.
+#>
+[CmdletBinding()]
+Param (
+  [Parameter(Mandatory=$True)]
+  [string] $ArtifactPath,
+  [Parameter(Mandatory=$True)]
+  [string] $TargetPath
+)
+
+. (Join-Path $PSScriptRoot "Helpers" git-helpers.ps1)
+
+function Get-ChangedServices {
+    Param (
+        [Parameter(Mandatory=$True)]
+        [string[]] $ChangedFiles
+    )
+    
+    $changedServices = $ChangedFiles | Foreach-Object { if ($_ -match "sdk/([a-zA-Z\-]+)") { $matches[1] } } | Sort-Object -Unique
```

Comment:
```suggestion
    $changedServices = $ChangedFiles | Foreach-Object { if ($_ -match "sdk/([^/]+)") { $matches[1] } } | Sort-Object -Unique
```
Might be easier to match anything other than the next directory separator. I cannot remember for sure but does the Get-ChangedFiles always come back with forward slashes from git on both windows and linux?

Reply:
I will confirm this and get back to you.

Reply:
I mean if not it just changes to `[^/\\]`

Reply:
Confirmed. This is the output for a windows agent:
![image](https://github.com/Azure/azure-sdk-tools/assets/45376673/2b022517-bf7e-42a5-a69f-8ff6e18a2f0c)

Identical.